source/deploying-with-kubernetes/kubernetes-conf.rst

```diff
@@ -0,0 +1,327 @@
+
+    - Having at least two Kubernetes nodes in order to meet the *podAntiAffinity* policy.
+
+Overview
+--------
+
+StateFulSet and Deployments Controllers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Like a *Deployment*, a *StatefulSet* manages Pods that are based on an identical container specification, but it maintains an identity attached to each of its pods. These pods are created from the same specification, but they are not interchangeable: each one has a persistent identifier maintained across any rescheduling.
+
+It is useful for stateful applications like databases that save the data to a persistent storage. The states of each Wazuh manager as well as Elasticsearch are desirable to maintain, so we declare them using StatefulSet to ensure that they maintain their states in every startup.
+
+Deployments are intended for stateless use and are quite lightweight and seem to be appropriate for Logstash, Kibana and Nginx, where it is not necessary to maintain the states.
+
+A **PersistentVolume (PV)** is a piece of storage in the cluster that has been provisioned. It is a resource in the cluster just like a node is a cluster resource. PVs are volume plugins like Volumes but have a lifecycle independent of any individual pod that uses the PV. This API object captures the details of the implementation of the storage, be that NFS, iSCSI, or a cloud-provider-specific storage system.
+Here, we use PVs to storage data of interest from both Wazuh and Elasticsearch. 
+They work in a similar way to docker volumes (vital for updates).
+
+.. note::
+    Users can read more information about the **Persistent volumes** in the `Kubernetes Official documentation. <https://kubernetes.io/docs/concepts/storage/persistent-volumes/>`_
+
+Pods
+^^^^
+
+.. note::
+    In this `link <https://github.com/wazuh/wazuh-docker>`_ you can check the data Wazuh uses in its docker deployments.
+
+**Wazuh master**
+
+This pod contains the master node of the Wazuh cluster. The master node centralizes and coordinates worker nodes, making sure the critical and required data is consistent across all nodes. The management is performed only in this node, so the agent registration service (authd) and the API are placed here.
+
+Details:
+    - Image: Docker Hub 'wazuh/wazuh:3.9.0_6.6.2'
+    - Controller: StatefulSet
+
+**Wazuh worker 0 / 1**
+
+These pods contain a worker node of the Wazuh cluster. They will receive the agent events.
+
+Details:
+    - Image: Docker Hub 'wazuh/wazuh:3.9.0_6.6.2'
+    - Controller: StatefulSet
+
+**Elasticsearch**
+
+Elasticsearch pod. It receives and stores alerts received from Logstash. No Elasticsearch cluster is supported yet.
+
+Details:
+    - Image: Docker Hub 'wazuh/wazuh-elasticsearch:3.9.0_6.6.2'
+    - Controller: StatefulSet
+
+**Logstash**
+
+Logstash pod. It receives the alerts from each Filebeat located in every Wazuh manager. Then, the alerts are sent to Elasticsearch.
```

Comment:
Logstash pod, it's listening to events from the Filebeat instances that are installed on every Wazuh manager node, then it sends all the events to Elasticsearch.